src/tree/serviceExplorer.ts

```diff
@@ -28,10 +34,14 @@ export class ServiceExplorer implements Disposable {
 
   private fsw: FileContentChangeNotifier;
 
+  private fswEnv: FileContentChangeNotifier;
+
   public constructor() {
     const treeDataProvider = new ServiceDataProvider();
     this.fsw = WatchUtil.watchFileForContextChange(kubeConfigFolder, 'config');
+    this.fswEnv = WatchUtil.watchFileForContextChange(kubeConfigFolderENV, 'kubeconfig');
```

Comment:
KUBECONFIG points to one or more files [1] . I do not think you should assume a name here.

https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#the-kubeconfig-environment-variable

Reply:
I didn't realize. Thanks for point that out.